django/utils/encoding.py

```diff
@@ -109,6 +110,43 @@ def smart_bytes(s, encoding='utf-8', strings_only=False, errors='strict'):
     return force_bytes(s, encoding, strings_only, errors)
 
 
+def convert_encodings(uri):
+    """Convert the %encodings back to their octet except the encoding for the '%' sign"""
+    _hexdig = '0123456789ABCDEFabcdef'
+    _hextochr = dict((a + b, chr(int(a + b, 16)))
+                 for a in _hexdig for b in _hexdig)
+    _asciire = re.compile('([\x00-\x7f]+)')
+
+    if isinstance(uri, (str, unicode)):
```

Comment:
To fix the `unicode` issue, use `six.string_types` instead.
https://pythonhosted.org/six/#six.string_types